doc/decisions/operation_sequences.md

```diff
@@ -0,0 +1,94 @@
+# KDB operation sequences
+
+## Problem
+
+The current public API allows various kind of sequences of `kdbGet` and `kdbSet` calls.
+There is no real guideline of allowed and/or prohibited sequences, other than `kdbGet` for a certain parent key must be called before `kdbSet` on the same key.
```

Comment:
Specifically, the current constraints are

https://github.com/ElektraInitiative/libelektra/blob/d1a8f828fea2ad77c612df62009764beb397d273/src/libs/elektra/kdb.c#L2194-L2196